internal/brokers/withexamples.go

```diff
@@ -0,0 +1,118 @@
+//go:build withexamplebroker
+
+package brokers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/ubuntu/authd/internal/brokers/examplebroker"
+	"github.com/ubuntu/authd/internal/log"
+)
+
+func useExampleBrokers() (string, error) {
+	busCleanup, err := startSystemBusMock()
+	if err != nil {
+		return "", err
+	}
+	log.Debugf(context.Background(), "Mock system bus started on %s\n", os.Getenv("DBUS_SYSTEM_BUS_ADDRESS"))
+
+	// Create the directory for the broker configuration files.
+	cfgPath, err := os.MkdirTemp(os.TempDir(), "examplebroker.d")
+	if err != nil {
+		busCleanup()
+		return "", err
+	}
+	cleanup := func() {
+		os.RemoveAll(cfgPath)
+		busCleanup()
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		if err = examplebroker.StartBus(ctx, cfgPath); err != nil {
+			fmt.Printf("Error starting example broker: %v\n", err)
+			cleanup()
+		}
+	}()
+	// Give some time for the broker to start
+	time.Sleep(time.Second)
+
+	go func() {
+		time.Sleep(time.Minute)
+		cancel()
+		<-done
+		cleanup()
+		fmt.Println("Stopped example broker")
+	}()
```

Comment:
This is definitely far from ideal, but I didn't have any other ideas on how to clean everything up (since SIGINT'ing the daemon would exit everything and give no time to cleanup the dbus mock and the examplebroker files).

Reply:
I think we should have a manager.Stop() which is a no-op in prod and does this cleanup here. Basically we will call it by defering after creating the manager in `func (a *App) serve(config daemonConfig) error {`.

WDYT?